openfermioncirq/optimization/black_box.py

```diff
@@ -93,3 +99,91 @@ def noise_bounds(self,
         return a value within the bounds.
         """
         return -numpy.inf, numpy.inf
+
+
+class StatefulBlackBox(BlackBox):
+    """A black box that keeps track of things.
```

Comment:
Maybe "A black box function with memory of evaluations" would be more descriptive?